new_app_setup.py

```diff
@@ -0,0 +1,266 @@
+
+    with open(config_filename, 'w') as outfile:
+        subprocess.call(terminal_command, stdout=outfile)
+
+    print("The config file for {} was created and "
+          "saved as {}".format(team_name, config_filename))
+    return config_filename
+
+
+def send_config_email_to_users(team_name, email_to, config_filename):
+    '''
+    Note: This code is going to change to reflect which email address we
+    would be sending this from. This successfully sends emails to GMAIL
+    and UOREGON accounts, I have not tried to send to any
+    other emails.
+    '''
+    # QUESTION: What email address do you want me to have these
+    # config files send from?
+    MVP_email_address = 'abkelley97@gmail.com'
+
+    # This line will be the actual one used to reflect the password
+    # for whatever email we send from.
+    # MVP_email_address_password = os.environ.get
+    #                               ('MVP_EMAIL_ADDRESS_PASSWORD')
+
+    # TEMP for my testing right now:
+    MVP_email_address_password = os.environ[str('PROF_EMAIL_PASS')]
+
+    email_contents_subject = ("Kubernetes config file for "
+                              "Team " + team_name + " at Hack For A Cause")
+
+    email_contents_body = ("Hi there,\nAttached to this email is your "
+                           "Kubernetes configuration file for your team " +
+                           team_name + ".\nThanks, \nMVP Stuido")
+
+    msg = MIMEMultipart()
+    msg['From'] = MVP_email_address
+    msg['Subject'] = email_contents_subject
+
+    msg.attach(MIMEText(email_contents_body, 'plain'))
+
+    attachment = open(config_filename, "rb")
+
+    # Allows us to upload the attachment
+    part = MIMEBase('application', 'octet-stream')
+    part.set_payload((attachment).read())
+    encoders.encode_base64(part)
+    part.add_header('Content-Disposition', "attachment; filename= " +
+                    config_filename)
+
+    msg.attach(part)  # for the attachment
+    text = msg.as_string()
+    server = smtplib.SMTP('smtp.gmail.com', 587)
+    server.starttls()
+    server.login(MVP_email_address, MVP_email_address_password)
+
+    if type(email_to) is list:
+        for user in email_to.split(','):
+            receiver_email_address = user
+            msg['To'] = receiver_email_address
+            server.sendmail(MVP_email_address, receiver_email_address, text)
+            print ("The config file for {} has successfully been sent "
+                   "to {}".format(team_name, user))
+
+    else:
+        receiver_email_address = email_to
+        msg['To'] = receiver_email_address
+        server.sendmail(MVP_email_address, receiver_email_address, text)
+        print ("The config file for {} has successfully been sent "
+               "to {}".format(team_name, email_to))
+
+    server.close()
+
+
+def add_users(filename, group_user, list_of_users):
+    '''
+    QUESTIONS: I am still unclear after looking into this a good amount how we
```

Comment:
You are on the right track. We can start by just creating a single ServiceAccount for the team. Then a Role within their namespace that has full access to every thing (you can use a *), then a RoleBinding that applies that new Role to the new ServiceAccount. Note, we don't want to use ClusterRole or ClusterRoleBinding as that would give them access to everything. If it helps we can go through doing this manually on Tuesday.

Example of a Role: https://kubernetes.io/docs/reference/access-authn-authz/rbac/#role-and-clusterrole
Example of a RoleBinding: https://kubernetes.io/docs/reference/access-authn-authz/rbac/#rolebinding-and-clusterrolebinding

Reply:
Service accounts: "User accounts are for humans. Service accounts are for processes, which run in pods." from https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#user-accounts-vs-service-accounts

From https://kubernetes.io/docs/reference/access-authn-authz/authentication/#users-in-kubernetes:

> All Kubernetes clusters have two categories of users: service accounts managed by Kubernetes, and normal users.

> Normal users are assumed to be managed by an outside, independent service. An admin distributing private keys, a user store like Keystone or Google Accounts, even a file with a list of usernames and passwords. In this regard, Kubernetes does not have objects which represent normal user accounts. Normal users cannot be added to a cluster through an API call.

> In contrast, service accounts are users managed by the Kubernetes API. They are bound to specific namespaces, and created automatically by the API server or manually through API calls. Service accounts are tied to a set of credentials stored as Secrets, which are mounted into pods allowing in-cluster processes to talk to the Kubernetes API.

In short - users should be added via `gcloud` as `gcloud` is the identity provider for our cluster. I think the relevant docs are here: https://cloud.google.com/kubernetes-engine/docs/how-to/iam

I **think** what we need to do is:

1. Use gcloud to create users in our organization with role `roles/container.developer`
2. Use `Role` and `RoleBinding` as @slively describes to limit the user to only their namespace

Reply:
Alright, so I am making the user as so:

        GROUP = 'mvpstudio'
        RESOURCE_NAME = team_name
        ROLE_ID = 'roles/container.developer'


        subprocess.check_call(['gcloud', GROUP, 'add'+'-'+'iam'+'-'+'policy'+'-'+'binding',
                RESOURCE_NAME, '--member', 'user'+':'+group_user, '--role', ROLE_ID])


and then I see there are options to make a Role and RoleBinding yaml, or to just have a command line do this with specific flags. 

Right now I am working with this:

      def create_role_yml(role_filename, team_name, group_user):
         """
         https://kubernetes.io/docs/reference/access-authn-authz/rbac/#role-and-clusterrole
         """
           yml_file_kubernetes_data = dict(

             apiVersion='rbac.authorization.k8s.io/v1',
             kind='Role',
               metadata=dict(
                 namespace=team_name,
                    name=group_user
               )
               rules=dict(
                   apiGroups=[""],
                    resourses=["pods"],
                    verbs=["get", "watch", "list"]
                 )
                )

           with open(role_filename, 'w') as outfile:
              yaml.dump(yml_file_kubernetes_data, outfile, default_flow_style=False)
   
          subprocess.check_call(['kubectl', 'create', 'role', role_filename])

       def create_rolebinding_yml(rolebinding_filename, team_name, group_user):

          yml_file_kubernetes_data = dict(

              apiVersion='rbac.authorization.k8s.io/v1',
             kind='RoleBinding',
                metadata=dict(
                  name=read-pods,
                  namespace=team_name,
                    )
             subjects=dict(
                  kind= User,
                  name=group_user,
                  apiGroup=rbac.authorization.k8s.io,
            )
             roleRef=dict(
                   kind= Role,
                   name= team_name,
                   apiGroup=rbac.authorization.k8s.io,
                    )
              )

         with open(role_filename, 'w') as outfile:
                  yaml.dump(yml_file_kubernetes_data, outfile, default_flow_style=False)

        subprocess.check_call(['kubectl', 'create', 'rolebinding', rolebinding_filename, '--namespace'+'='+team_name])

This isn't completely finished, but I want to make sure I am not doing it all wrong.

Reply:
that looks basically right. As per other comments I think `{}` is more common/Python-ic than `dict(...)` but it doesn't really matter much.

Also, isn't this:

```
'add'+'-'+'iam'+'-'+'policy'+'-'+'binding'
```

a long way to say this:

```
'add-iam-policy-binding'
```

Reply:
I am still having issues with adding the users, but since we are focuing on the namespace and resourcequota for this PR, I will include that in a new branch.